python/meep.i

```diff
@@ -793,13 +805,7 @@ void _get_gradient(PyObject *grad, PyObject *fields_a, PyObject *fields_f, PyObj
     if (!PyArray_Check(pao_freqs)) meep::abort("frequencies parameter must be numpy array.");
     if (!PyArray_ISCARRAY(pao_freqs)) meep::abort("Numpy fields array must be C-style contiguous.");
     meep::realnum* frequencies_c = (meep::realnum *)PyArray_DATA(pao_freqs);
-
-    // get the proper dimensions of the fields array
-    if (PyArray_NDIM(pao_fields_a) !=5) {meep::abort("Fields array must have 5 dimensions.");}
```

Comment:
Do we still need a `PyArray_NDIM` check?

Reply:
No, we don't

Reply:
So the code requires `PyArray_NDIM(pao_fields_a) == 1`, then.

Reply:
Yes should be fixed now